### DISCUSSION TITLE
SUMO multi-clients connection

### DISCUSSION THREAD

**Discussion by chengyuan0124** [eclipse/mosaic#293] [Q&A] [answered]
Hello there,

I am working on a project that requires me to use both MOSAIC and a self-developed script to connect to SUMO. However, as MOSAIC already has a Traci client to connect to SUMO and my script also requires a Traci client to set up the connection, I may need to have two clients and use the [setOrder](https://sumo.dlr.de/docs/TraCI/Control-related_commands.html#command_0x03_setorder) function for each Traci client connection.

Does any `setOrder` configuration exists in the MOSAIC currently? Or it has to develop function myself?

**Comment by schwepmo** [comment, marked as answer]
Hello,
I'm not familiar with using multiple traci clients. However, it is possible to set arbitrary parameter for the sumo call using the `additionalSumoParameters` in the *sumo_config.json* of your scenario. If that doesn't help you could try to implement a solution yourself as it takes some time to internally evaluate the need and effort of new features.

Edit: I just saw that this is an actual traci-call, we don't implement this yet and you can try coming up with a solution yourself. If you manage to resolve this feel free to open a Pull Request and we will look into it!
